pkg/services/chgm/chgm.go

```diff
@@ -50,32 +58,117 @@ type Service interface {
+	DeleteLimitedSupportReasons(ls ocm.LimitedSupportReason, clusterID string) error
 	// PD
-	AddNote(incidentID string, noteContent string) error
-	ExtractServiceIDFromPayload(payloadFilePath string, reader pagerduty.FileReader) (string, error)
-	CreateNewAlert(description string, details interface{}, serviceID string) error
-	MoveToEscalationPolicy(incidentID string, escalationPolicyID string) error
-	GetEscalationPolicy() string
-	GetSilentPolicy() string
+	SilenceAlert(notes string) error
+	AddNote(notes string) error
+	CreateNewAlert(newAlert pagerduty.NewAlert, serviceID string) error
+	GetServiceID() string
+	UpdateAndEscalateAlert(notes string) error
 }
 
-// Client is an implementation of the Interface, and adds functionality above it
-// to create 'Client' you can use a mock, or fill it with
-// this differs from the ChgmProvider as it makes sure you are intentional when using functions.
-// if I am missing a func I will copy it from the corresponding package to the interface instead of
-// having a function change break my code.
-// TODO: decide if the Client should be the ChgmProvider
+// Client for the chgm investigation
 type Client struct {
 	Service
-	cluster *v1.Cluster
-	cd      *hivev1.ClusterDeployment
+	Cluster           *v1.Cluster
+	ClusterDeployment *hivev1.ClusterDeployment
+	*investigation.Client
+}
+
+// Triggered will analyse chgm incidents that are newly created
+func (c *Client) Triggered() error {
+	res, err := c.investigateStoppedInstances()
+	if err != nil {
+		return c.UpdateAndEscalateAlert(fmt.Sprintf("InvestigateInstances failed: %s\n", err.Error()))
+	}
+
+	fmt.Printf("the investigation returned %#v\n", res)
+
+	lsExists, err := c.LimitedSupportReasonsExist(c.Cluster.ID())
+	if err != nil {
+		return c.UpdateAndEscalateAlert(fmt.Errorf("failed to determine if limited support reason already exists: %w", err).Error())
+	}
+
+	// if lsExists, silence alert and add investigation to notes
+	if lsExists {
+		fmt.Println("Unrelated limited support reason present on cluster, silencing")
+		return c.SilenceAlert(res.String() + "Unrelated limited support reason present on cluster, silenced.")
+	}
+
+	if res.UserAuthorized {
+		fmt.Println("The node shutdown was not the customer. Should alert SRE")
+		return c.UpdateAndEscalateAlert(res.String())
+	}
+
+	// The node shutdown was the customer
+	// Put into limited support, silence and update incident notes
+	fmt.Printf("Sending limited support reason: %s\n", chgmLimitedSupport.Summary)
+	return utils.Retry(3, time.Second*2, func() error {
+		return c.PostLimitedSupport(res.String())
+	})
+}
+
+// Resolved will take appropriate action against a cluster whose CHGM incident has resolved.
+func (c *Client) Resolved() error {
+	res, err := c.investigateStartedInstances()
+
+	// The investigation encountered an error & never completed - alert Primary and report the error, retrying as many times as necessary to escalate the issue
```

Comment:
**issue:** We went from the logic being "retry endlessly" to "retry 3 times". I think we can allow a higher retry count for all the retries (maybe 10?). This could be a global variable as well.

The comment should be changed as well.

As discussed though, in the worst case we do not remove LS and the customer will have to go the normal route of opening a CS ticket. That's not dramatic.